core/base/mtx_io.cpp

```diff
@@ -78,18 +78,30 @@ class mtx_reader {
     {
         auto parsed_header = this->read_header(is);
         std::istringstream dimensions_stream(parsed_header.dimensions_line);
-        auto data = parsed_header.layout_reader->read_data(
-            dimensions_stream, is, parsed_header.entry_reader,
-            parsed_header.layout_reader_modifier);
+        auto data = parsed_header.layout->read_data(
+            dimensions_stream, is, parsed_header.entry, parsed_header.modifier);
         data.ensure_row_major_order();
         return data;
     }
 
+    void write(std::ostream &os, const matrix_data<ValueType, IndexType> &data,
+               const std::string &header) const
+    {
+        std::istringstream header_stream(header);
+        auto parsed_header = this->read_description_line(header_stream);
+        CHECK_STREAM(os << header,
+                     "error when writing the matrix market header");
+        parsed_header.layout->write_data(os, data, parsed_header.entry,
+                                         parsed_header.modifier);
+    }
+
 private:
     // entry format hierarchy provides algorithms for reading a single entry of
```

Comment:
nit: Technically it's not only reading now. Also in most other similar comments.

Reply:
good catch, that's why comments are evil :imp: